README.md

```diff
@@ -50,6 +50,18 @@ git clone https://github.com/kaitai-io/kaitai_struct_visualizer
 ksv <binary-file> <ksy-file>... | <rb-file>
 ```
 
+### Using Docker image
+
+Running image from DockerHub:
+```shell
+docker run -v "$(pwd):/share":Z -it yakshaveinc/ksv <binary-file> <ksy-file>
```

Comment:
Mentioning it in README means that this Docker image becomes an officially supported one, and we'll maintain it from now on. That's ok, but that requires, at bare minimum:

- [x] Creating [kaitai org](https://hub.docker.com/orgs/kaitai) at Docker hub (or choosing to use Docker repos provided by GitHub)
- [ ] Putting that under kaitai org, accessible by more than 1 person
- [ ] Documenting release procedure at https://github.com/kaitai-io/kaitai_struct_doc/blob/master/developers.adoc

Ideally, we'd want this to be automated, so creating a release would be something like "run official builds" -> double check manually -> "run official release".

Reply:
@GreyCat if you already set DOCKER_USER and DOCKER_PASS then like described here https://docs.travis-ci.com/user/docker/#pushing-a-docker-image-to-a-registry then it is not a problem

Reply:
@GreyCat 

> Putting that under kaitai org, accessible by more than 1 person

I created https://hub.docker.com/repository/docker/kaitai/kaitai-struct-visualizer. However, I'm not sure what versioning scheme we should choose for this package. In fact, that's all which matters here - there's no problem with setting up a CI build automatically publishing the Docker image and gem after pushing to `master`.

The existing gem on RubyGems ([kaitai-struct-visualizer](https://rubygems.org/gems/kaitai-struct-visualizer/)) has the series 0.3, 0.4, 0.5, 0.7, which seems to correlate to the KSC versions. But that doesn't allow releasing new versions of `ksv` more often than the versions of KSC without having the version numbers diverged.

We can switch to some `-SNAPSHOT` versioning for `ksv`, but I'm not really a fan of it. We should rather adopt proper semantic versioning for the whole Kaitai Struct project.

Reply:
@generalmimon how about `latest` for now and everything else later for manual releases?

Reply:
> https://hub.docker.com/repository/docker/kaitai/kaitai-struct-visualizer

I would create shorter name for that, llike `kaitai/vis`

Reply:
> > https://hub.docker.com/repository/docker/kaitai/kaitai-struct-visualizer
> 
> I would create shorter name for that, llike `kaitai/vis` 

Hmm, we should keep the naming consistent, so I would go either for `kaitai/ksv` or perhaps `kaitai/visualizer`. Agreed that `kaitai/kaitai-struct-visualizer` is probably too long for manual typing.

> @generalmimon how about `latest` for now and everything else later for manual releases?

OK, I haven't realized that you can overwrite existing tags like `latest`, and versions don't need to have unique tags. Do I understand it correctly that there will be always just one image version at the time, or perhaps you can see all published versions over time accessible by some image hash (analogously to Git commit hashes)?

Reply:
> Agreed that kaitai/kaitai-struct-visualizer is probably too long for manual typing.

`kaitai/ksv` would be good for now. Although it ships `ksc` as well, which will be confusing as people will start using this Docker image for invoking `ksc` (I do). )

`latest` is always overwritten, right. It is possible to pull image be the previous hash https://docs.docker.com/engine/reference/commandline/pull/#/pull-an-image-by-digest-immutable-identifier Not sure if old images are retained.

Reply:
Hmm, Docker Hub doesn't allow renaming repositories - good to know. I'll delete the empty repository and create a new one then.

Reply:
https://hub.docker.com/repository/docker/kaitai/ksv

Reply:
@generalmimon if Travis is enabled, I can commit auto-publishing on every commit, or GitHub Actions f you prefer.

Reply:
I'd prefer GitHub Actions. Note that we're currently moving away from Travis (https://github.com/kaitai-io/kaitai_struct/issues/824), and GH Actions looks like a solid replacement. It has (unsurprisingly) the best integration with GitHub out-of-the-box :) And as @GreyCat said somewhere, using GH Actions means one less external service we have to care about.

Reply:
GitHub doesn't preserve build logs - https://github.blog/changelog/2020-10-08-github-actions-ability-to-change-retention-days-for-artifacts-and-logs/

Reply:
Build logs are essential for old bug reports. My "hello world" GitHub Action logs are [already gone](https://github.com/yakshaveinc/linux/runs/687817737) after 6 months, and on Travis CI I can see at least 2 years old logs https://travis-ci.org/github/kaitai-io/kaitai_struct/builds/485169049

Reply:
@abitrolly:
> GitHub doesn't preserve build logs

Indeed, it keeps them for a maximum of 90 days (3 months). That's a bit unfortunate, but not critical.

@abitrolly:
> Build logs are essential for old bug reports. My "hello world" GitHub Action logs are [already gone](https://github.com/yakshaveinc/linux/runs/687817737) after 6 months, and on Travis CI I can see at least 2 years old logs https://travis-ci.org/github/kaitai-io/kaitai_struct/builds/485169049

Hmm, old bug reports... It probably means bugs that were present in some old versions, but are no longer present in the current one. I don't really understand what you mean by this. At the moment, we only fix bugs in the latest version, we do not update old releases using patch versions like `0.6.1` for example (that would fix some bug in KS 0.6). When we release a new version, all previous ones are effectively EOL, and there is IMHO no need to dig into the old bug reports.

Of course, this may change when we switch to a proper semantic versioning, but it is the current state of things.

Reply:
@abitrolly https://github.com/kaitai-io/kaitai_struct/issues/824#issuecomment-730686323

Reply:
I find old build logs necessaty. I spent too much time giving direct link to certain lines in log files, so that people who came with the same problem could have the prooflink. That's just my personal preference.

As for adding the GitHub Action I see that Docker now promotes new `bulldx` system as a right way to build containers and I need to study that before using. You may go ahead and just apply official action https://github.com/marketplace/actions/build-and-push-docker-images

Reply:
#46 adds actions to build and push to Docker Hub automatically.